apps/src/templates/teacherDashboard/AddInitialStudentsView.jsx

```diff
@@ -37,87 +41,118 @@ class AddInitialStudentsView extends Component {
           </Heading3>
         )}
         <CardContainer>
-          <PictureLoginCard/>
-          <WordLoginCard/>
-          <EmailLoginCard/>
+          <PictureLoginCard
+            onClick={handleLoginChoice}
+          />
+          <WordLoginCard
+            onClick={handleLoginChoice}
+          />
+          <EmailLoginCard
+            onClick={handleLoginChoice}
+          />
         </CardContainer>
         {anyThirdParty && (
           <div>
             <Heading3>
               {i18n.addStudentsSyncThirdParty()}
             </Heading3>
             <CardContainer>
-              {googleClassroom && <GoogleClassroomCard/>}
-              {microsoftClassroom && <MicrosoftClassroomCard/>}
-              {clever && <CleverCard/>}
+              {googleClassroom &&
+              <GoogleClassroomCard
+                onClick={handleLoginChoice}
+              />}
+              {microsoftClassroom &&
+              <MicrosoftClassroomCard
+                onClick={handleLoginChoice}
+              />}
+              {clever &&
+              <CleverCard
+                onClick={handleLoginChoice}
+              />}
             </CardContainer>
           </div>
         )}
+        <DialogFooter>
+          <ProgressButton
+            onClick={handleCancel}
+            text="Cancel"
+            size={ProgressButton.ButtonSize.large}
+            color={ProgressButton.ButtonColor.gray}
+          />
+        </DialogFooter>
       </div>
     );
   }
 }
 export default AddInitialStudentsView;
 
-const PictureLoginCard = () => (
+const PictureLoginCard = (props) => (
   <LoginTypeCard
     title={i18n.loginTypePicture()}
     subtitle={i18n.loginTypePictureAgeGroup()}
     description={i18n.loginTypePictureDescription()}
     buttonText={i18n.loginTypePictureButton()}
-    link="#"
     isRtl={false}
+    onClick={props.onClick.bind(this, 'picture')}
```

Comment:
I dont feel super strongly, but I think I prefer using arrow functions to express this, i.e. `onClick={() => props.onClick('picture')`}